dispatch/queue.h

```diff
@@ -1346,6 +1346,16 @@ dispatch_assert_queue_not(dispatch_queue_t queue)
 #define dispatch_assert_queue_not_debug(q) dispatch_assert_queue_not(q)
 #endif
 
+/*!
+ * @handler dispatch_thread_detach_handler
+ *
+ * Hook to be able to detach threads before they exit from the Java JVM.
+ * If JNI has been used on a thread on Android it needs to have been
+ * "detached" before the thread exits or the application will crash.
+ */
+DISPATCH_EXPORT
+void (*dispatch_thread_detach_handler)();
```

Comment:
1) this should go to queue_internal.h
2) this should be `(void)`
3) this should only be enabled on android